src/app/cli/src/cli_entrypoint/mina_cli_entrypoint.ml

```diff
@@ -913,12 +913,27 @@ let setup_daemon logger =
             client_trustlist
       in
       Stream.iter
-        (Async.Scheduler.long_cycles
+        (Async_kernel.Async_kernel_scheduler.(long_cycles_with_context (t ()))
            ~at_least:(sec 0.5 |> Time_ns.Span.of_span_float_round_nearest))
-        ~f:(fun span ->
+        ~f:(fun (span, ctx) ->
           let secs = Time_ns.Span.to_sec span in
+          let rec get_monitors accum monitor =
+            match Async_kernel.Monitor.parent monitor with
+            | None ->
+                List.rev accum
+            | Some parent ->
+                get_monitors (parent :: accum) parent
+          in
+          let monitors = get_monitors [ctx.monitor] ctx.monitor in
+          let monitor_infos =
+            List.map monitors ~f:(fun monitor ->
+                `String
```

Comment:
It might be more readable if you use `Error_json.sexp_to_json` here?

Reply:
I realize now that there's an existing use of the `with_context` call, I'm going to move the `monitors` log there.

I created a PR against `async_kernel` to expose the `here` field. But maybe your suggestion is simpler.

Reply:
Oops, that other `with_context` is for jobs, not cycles.